pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -239,6 +239,18 @@ def _map_tool_result_part(
                 )
             else:
                 assert_never(resource)
+        elif isinstance(part, mcp_types.ResourceLink):
```

Comment:
Can you please reduce the duplication between this new branch and the one for `isinstance(part, mcp_types.EmbeddedResource)`?